### ISSUE TITLE
Provide `reservedWord` parser?

### ISSUE THREAD

**Issue by Anrock** [mrkkrp/megaparsec#334]
Hi, i'm mainly using Megaparsec for toy languages and was wondering why it doesn't provide some kind of parser for reserved words? I think it would nicely fit into lexer module.

Tutorial even provides a example implementation:
```
rword :: String -> Parser ()
rword w = (lexeme . try) (string w *> notFollowedBy alphaNumChar)

rws :: [String] -- list of reserved words
rws = ["if","then","else","while","do","skip","true","false","not","and","or"]
```

Can we get that into library so i could stop copypasting it everywhere?

**Comment by mrkkrp** [comment]
Well, the amount of code you copy-paste is not that big, and different languages have different syntax with respect to what is considered identifier. So even if we add it, it'll have to take at least two parameters already, so it's really easier to simply "inline" them and have your own definition. My opinion is that this is just too-specific to a given language and won't help much if we add it to the library.

**Comment by Anrock** [comment]
@mrkkrp thanks for quick response, however i have to disagree - your point is applicable to any function in Text.Megaparsec.Lexer. All of them take more than two parameters and (arguably) all of them are shorter than proposed parser and some of them more or less language-specific. 
So i don't see why proposed function doesn't fit here.

**Comment by Anrock** [comment]
> Is your rword significantly different from symbol from one of the Lexer modules?

Not really. `symbol "reserved"` would succesfully parse `"reservedabc"` while `rword` would fail due to unexpected `"abc"` piece.

If i understand correctly - you and your coallegue are mixing lexing and parsing together, while i prefer to separate these phases - write dumb lexer to get `[Token]` first and then write a parser that operates on those `Token`s instead of raw `Char`s.  I find this approach easier overall (at least for my toy languages) - trivial lexer and easy parser instead of single phase of medium difficulty.

So basically `rword` would improve Megaparsec for my use case by providing common combinator, just like already provided `lexeme`, `space` and others.

**Comment by mrkkrp** [comment]
I'm going to close this one. I'm not a fan of this new function, but if you absolutely want it, I think we could merge a PR implementing it.
